modules/apps/portlet-configuration/portlet-configuration-css-web/package.json

```diff
@@ -1,7 +1,8 @@
 {
 	"dependencies": {
 		"@clayui/color-picker": "3.25.1",
-		"react": "16.12.0"
+		"react": "16.12.0",
+		"tinycolor2": "1.4.1"
```

Comment:
I'm pretty sure we don't want to introduce any external dependencies, like this `tinycolor2`. Especially, correct me if I'm wrong, inside `portlet-configuration` which I assume by it's name influences all portlets inside DXP. If anything we should probably do this color manipulation closer to the problem, inside the Clay component perhaps.

Reply:
`tinycolor2` is already coming in [via Clay](https://github.com/liferay/clay/blob/6ee32869e2f42a392edfc6ad0d9f9d746891d685/packages/clay-color-picker/src/index.tsx#L11) but we should make sure that we don't introduce a duplicate/redundant version here, because it's actually [not that "tiny" at all](https://bundlephobia.com/result?p=tinycolor2@1.4.1):

<img width="466" alt="Screenshot 2021-05-06 -115542-XKeUIU9m@2x" src="https://user-images.githubusercontent.com/7074/117279449-02af4100-ae62-11eb-87fc-4c22ba207724.png">

5KB gzipped is pretty huge for what it's actually doing...

So 1.4.1 is the version currently in `liferay-portal` via Clay right now, but declaring it independently like this here could potentially lead us to have duplicates in the future (if they fall out of sync). We can either do nothing about this and just be aware that it might be something we have to deal with in the future, or try to come up with a local function that does exactly what we need and no more, or put something in place to ensure that we are _guaranteed_ to only ever have one version (frankly, I don't think it's worth the effort, so I wouldn't do this last option).

Reply:
Yeah I think we should avoid bringing in this dependency directly to DXP. I think the best bet would be to request this type of util to get exposed from `@clayui/color-picker` and that way we can unify how devs consume it. We may also want to provide over color centric utils from color picker as well in the future

Reply:
And if this PR is urgent, I am fine with creating our own util as a placeholder for now and then we can create a Clay issue and make note to update this code when the next clay release is

Reply:
@bryceosterhaus maybe it would not be better to make the `value` also have the `#`, we sort of [remove the `# `when it exists](https://github.com/liferay/clay/blob/master/packages/clay-color-picker/src/index.tsx#L160-L162) within the value so we have an internal security that the `#` will not hinder the business logic and maybe we add the `#` in the call back to `onValueChange`. What do you think? Just wishing I didn't have to expose a utility.

Reply:
@matuzalemsteles we could use a ref to track if the initial value includes a hash or not. If it does include a hash, then we call onChange with a hash.

I am fine including it or not, we just need to be sure not to change its functionality that might break existing code that relies on having the hash or not.

Reply:
@bryceosterhaus yeah we would have to check the use cases in our codebase that expect the value without the hash, I think if someone is expecting this behavior we can add an API to return the value with the hash and we can make this the default behavior in a next major version.